src/filter.ts

```diff
@@ -32,11 +37,24 @@ export interface RangeFilter {
    * @maxItems 2
    * @minItems 2
    */
-  range: any;
+  range?: any;
+  lt?: number;
+  lte?: number;
+  gt?: number;
+  gte?: number;
+
+  negate?: boolean;
 }
 
 export function isRangeFilter(filter: any): filter is RangeFilter {
-  return filter && !!filter.field && isArray(filter.range) && filter.range.length === 2;
+  if (filter && !!filter.field) {
+    if (isArray(filter.range) && filter.range.length === 2) {
+      return true;
+    } else if (XOR(filter.gt !==undefined, filter.gte!==undefined) || XOR(filter.lt!==undefined, filter.lte!==undefined) ) {
```

Comment:
I don't think we need XOR here.    If users want to stupidly combine  `gt` and `gte`.  That should mean we `&&` the two?  

Pro-tip:    `a!==b` is the way to do XOR if you need in other occasion.